Vizzuality/heco-invest
# Dashboard favorites

## Description

**In this PR**  

- `BadgeNavigation` changes:  
  _The new `DashboardFavoritesLayout` uses this component as its sidebar's navigation_
  - Added a new `simple` theme  
  - Added the `badgePosition` prop so we can choose the position of the (number) badges  
  - Added the `orientation` prop, so we can display the navigation both horizontally and vertically  
- Added a `CardHoverToDelete` component: 
  _A component we can use to wrap `ProfileCard` and `ProjectCard` components in, in order to display the "Delete" icon when hovering the card._
- `account/favoritesService` service:  
  - Added services & hooks to retrieve the list of favorited: 
    - Projects
    - Project developers  
    - Investors 
- Other services:  
  - Favorite projects, project developers, investors query invalidation  
    _When the user favorites/un-favorites projects, project developers, investors_
- `Enums` changes: 
  - Added `Paths` for the dashboard favorites' pages 
  - Added `Queries` for the queries used to fetch the lists of favorited projects, project developers, and  investors.
- `DashboardLayout` changes:  
  - Added `sidebar` support  
    _Needed for the dashboard/favorites views._
- Added `DashboardFavoritesLayout`  
  _This layout is mostly a wrapper around the `DashboardLayout`. It adds the sidebar and is responsible for fetching the list of favorited projects, project developers, ~open calls~ to display on the navigation badges. The logic is similar to how we fetch data in the `Discover` layout._ 
- Added `dashboard/favorites/*` pages  
  - Projects
    _List of favorited projects_
  - Project developers
    _List of favorited project developers_
  - Investors
    _List of favorited investors_
  - Open calls
    _Placeholder only, given we don't have open calls implemented yet._

**Notes:** 
  - `Pagination` has not been implemented. It'll be addressed in [LET-853](https://vizzuality.atlassian.net/browse/LET-853)
  - Removing Projects, Project developers and Investors from the favorites will be addressed in [LET-854](https://vizzuality.atlassian.net/browse/LET-854), [LET-855](https://vizzuality.atlassian.net/browse/LET-854), [LET-855](https://vizzuality.atlassian.net/browse/LET-856)

## Testing instructions

Verify that, in the Dashboard:  
  - The user can see a list of favorited Projects, Project developers, investors  
  - The lists are correct  
  - The open calls page displays a message stating the user doesn't have favorited open calls yet  
  - Hovering over a Project, Project developer or investor card shows the "delete" icon  
  - The navigation works correctly 

## Tracking

[LET-613 | FE - Adapt BadgeNavigation component to be able to also display items vertically](https://vizzuality.atlassian.net/browse/LET-613)
[LET-614 | FE - DashboardLayout to support sidebars  ](https://vizzuality.atlassian.net/browse/LET-614)
[LET-840 | FE - Add DashboardFavourites layout](https://vizzuality.atlassian.net/browse/LET-840)

[LET-615 | FE - Signed in PDs and Investors can see a list of favorited Projects](https://vizzuality.atlassian.net/browse/LET-615)
[LET-616 | FE - Signed in PDs and Investors can see a list of favorited Project developers](https://vizzuality.atlassian.net/browse/LET-616)
[LET-617 | FE - Signed in PDs and Investors can see a list of favorited Investors](https://vizzuality.atlassian.net/browse/LET-617)

## Screen recording 

**Before adding the delete buttons** 

https://user-images.githubusercontent.com/6273795/182606183-26d8b724-005f-458c-a69e-d28a9119c7fa.mp4

**With the Delete buttons**  

https://user-images.githubusercontent.com/6273795/182624829-b3fea743-96cc-4f06-8580-0cac219dede8.mp4
